### PR TITLE
feat!: updated boojum and nightly rust compiler

### PR DESCRIPTION
**Pull request by mm-zk**
## What ❔

* Updated boojum version, which allows us to update nightly rust compiler.

## Why ❔

* Our rust nightly is quite old (around 1 year)

### CODE REVIEW

**Comment by tomg10** [comment]
I believe you should also bump nightly version in zk-environment

**Review by popzxc — COMMENTED** [review]
It comes with a bunch of clippy issues (as clippy is updated too) -- unfortunately, it has to be fixed as well.

**Comment by popzxc** [comment]
Huh, I guess generic Poseidon implementation does have an impact on performance 👀

**Review by itegulov — APPROVED** [review]
I think this should also allow us to delete all false positive occurrences of `#[allow(clippy::needless_pass_by_ref_mut)]` in the code. Not a big deal if you don't want to do it in this PR though.

File: core/lib/dal/src/events_dal.rs [hunk 1]
```diff
@@ -84,7 +84,7 @@ impl EventsDal<'_, '_> {
                 write_str!(
                     &mut buffer,
                     r"\\x{topic0:x}|\\x{topic1:x}|\\x{topic2:x}|\\x{topic3:x}|",
-                    topic0 = EventTopic(event.indexed_topics.get(0)),
+                    topic0 = EventTopic(event.indexed_topics.first()),
                     topic1 = EventTopic(event.indexed_topics.get(1)),
                     topic2 = EventTopic(event.indexed_topics.get(2)),
                     topic3 = EventTopic(event.indexed_topics.get(3))
```
Comment (popzxc) [inline core/lib/dal/src/events_dal.rs:90]: Damn clippy, you're ruining all the aesthetics!

**Review by popzxc — DISMISSED** [review]

File: core/lib/dal/src/events_dal.rs (hunk 1 above)
Comment (mm-zk) [reply core/lib/dal/src/events_dal.rs:90]: yeah...

File: .dockerignore [hunk 2]
```diff
@@ -46,3 +46,4 @@ contracts/.git
 !etc/env/dev.toml
 !etc/env/consensus_secrets.yaml
 !etc/env/consensus_config.yaml
+!rust-toolchain
```
Comment (hatemosphere) [inline .dockerignore:49]: please re-add newline

**Review by hatemosphere — CHANGES_REQUESTED** [review]

File: .dockerignore (hunk 2 above)
Comment (mm-zk) [reply .dockerignore]: done

**Review by hatemosphere — APPROVED** [review]

**Review by popzxc — APPROVED** [review]

**Pull request merged**